src/main/java/net/minecraftforge/eventbus/LMFListenerFactory.java

```diff
@@ -0,0 +1,62 @@
+package net.minecraftforge.eventbus;
+
+import net.minecraftforge.eventbus.api.Event;
+import net.minecraftforge.eventbus.api.IEventListener;
+
+import java.lang.invoke.LambdaMetafactory;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+
+public class LMFListenerFactory implements IEventListenerFactory {
+    private static final MethodHandles.Lookup IMPL_LOOKUP;
+
+    static {
+        try {
+            var hackfield = MethodHandles.Lookup.class.getDeclaredField("IMPL_LOOKUP");
+            hackfield.setAccessible(true);
+            IMPL_LOOKUP = (MethodHandles.Lookup) hackfield.get(null);
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to access IMPL_LOOKUP...");
+        }
+    }
+
+    private static final MethodType LISTENER_INVOKE = MethodType.methodType(void.class, Event.class);
+
+    @Override
+    public IEventListener create(Method callback, Object target) {
+        try {
+            var callbackClass = callback.getDeclaringClass();
+
+            // We need to read the other module to perform java.lang.invoke operations with it
+            getClass().getModule().addReads(callbackClass.getModule());
```

Comment:
This is unnecessary with `IMPL_LOOKUP`

Reply:
I am only using the `IMPL_LOOKUP` to create a private lookup in the right class, so this is still needed unless we decide to use the `IMPL_LOOKUP` for everything (which I'd prefer to avoid).

Reply:
You're using the `IMPL_LOOKUP#in` lookup for everything so it should not be needed. LMF itself doesn't check the caller class.